src/lib.rs

```diff
@@ -246,18 +105,17 @@ impl TsunamiBuilder {
         self.max_wait = Some(t);
     }
 
-    /// Set the maxium lifetime of spawned spot instances.
+    /// Set the maxium lifetime of spawned instances, if applicable for the provider.
     ///
     /// EC2 spot instances are normally subject to termination at any point. This library instead
-    /// uses [defined
-    /// duration](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html#fixed-duration-spot-instances)
+    /// uses [defined duration](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html#fixed-duration-spot-instances)
     /// instances, which cost slightly more, but are never prematurely terminated. The lifetime of
     /// such instances must be declared in advance (1-6 hours), and can be changed with this
     /// method.
     ///
-    /// The default duration is 1 hour.
-    pub fn set_max_duration(&mut self, hours: u8) {
-        self.max_duration = hours as i64 * 60;
+    /// The default duration is 6 hours.
+    pub fn set_max_duration(&mut self, hours: u64) {
```

Comment:
You _could_ add a marker trait here along the lines of `SupportsDefinedDuration`, and then here add a bound:
```rust
where L: SupportsDefinedDuration
```
So that people don't accidentally use this where not supported.